examples/decode_progressive.cc

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This C++ example decodes a JPEG XL image progressively (input bytes are passed
+// in chunks). The example outputs the intermediate steps to PNG files.
+
+#include <limits.h>
+#include <lodepng.h>
```

Comment:
I would avoid the extra complexity in the example of writing png files.

Reply:
It's the easiest way to do RGBA output to easily viewable files though.

Reply:
The purpose of this example is to show the API usage; not to create useful tools.

Reply:
Sure, but outputting to 8-bit RGBA is nice to have as an additional example besides the one we already have (the one that outputs to float).

Reply:
If you install libjxl from a binary distribution and want to try out an example usage, you would need to also install lodepng source/dev files to save PNGs, which would be a blocker for trying out the example.

I don't think that passing num_channels=4 and JXL_TYPE_UINT8 makes much of a difference in the example source code.

Again, the goal of the examples/ directory is to show the API usage. The less lines you can use for that the better.